app/views/site/_study_visualize.html.erb

```diff
@@ -123,25 +123,20 @@
         $('#search_cluster').val(newCluster);
         $('#gene_set_cluster').val(newCluster);
         // grab currently selected annotation
-        var currAnnot = $('#annotation').val();
         var currSubsample = $('#subsample').val();
         // get new annotation options and re-render
         $.ajax({
             url: "<%= get_new_annotations_path(accession: @study.accession, study_name: @study.url_safe_name)%>?cluster=" + encodeURIComponent(newCluster) + '&subsample=' + encodeURIComponent(currSubsample),
             method: 'GET',
             dataType: 'script',
-            success: function (data) {
-                // parse response as a string and see if currently selected annotation exists in new annotations
-                if (data.indexOf(currAnnot) >= 0) {
-                    $('#annotation').val(currAnnot);
-                }
-                $(document).ready(function () {
+            complete: function (jqXHR, textStatus) {
+                if (textStatus === 'success') {
                     // since we now have new annotations, we need to set them in the search form for persistence
```

Comment:
that's interesting that the document.ready used to be there, presumably to solve a race condition where this ajax could complete before the html page finished loading.  It looks like that's no longer applicable

Reply:
True, but `complete` makes it even more explicit for the client to wait until all DOM updates have completed.  Every now and then I've seen some flakiness in UI tests with cluster/violin plot rendering and I'm thinking that this might be an underlying cause.